### PR TITLE
Add basic File model and use on documents page

### PR DESCRIPTION
**Pull request by KevinEtchells**
## Context

We're currently using dummy data on the documents page. This update now uses the Files model defined from the original Redbox app.

## Changes proposed in this pull request

* Bringing in the Files model from /redbox/models/file.py (this is not complete, I've just brought in and added what I need for now and added a comment)
* Allow for this to be used on the documents page

## Guidance to review

* Check the `/documents` page looks like this:
image

* Please check the File model I've created

### CODE REVIEW

**Comment by KevinEtchells** [comment]
Any help with the failing checks would be appreciated!

**Comment by KevinEtchells** [comment]
Yay, all checks now passing.

File: django_app/redbox_app/redbox_core/models.py [hunk 1]
```diff
@@ -31,3 +31,28 @@ class User(BaseUser, UUIDPrimaryKeyBase):
...
+    path = models.TextField()
+    name = models.TextField()
+    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, default=ProcessingStatusEnum.uploaded)
+
+    def get_processing_text(self) -> str:
+        processing_status_list = list(ProcessingStatusEnum)
+        stage = processing_status_list.index(self.processing_status)
+        if stage == len(processing_status_list) - 1:
+            return self.processing_status
+        else:
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/models.py:57]: nitpick: you dont need this `else` statement

File: django_app/redbox_app/redbox_core/models.py [hunk 2]
```diff
@@ -31,3 +31,28 @@ class User(BaseUser, UUIDPrimaryKeyBase):
...
+            return str(stage + 1) + "/" + str(len(processing_status_list) - 1) + " " + self.processing_status
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/models.py:58]:
```suggestion
            return f"{stage + 1}/{len(processing_status_list) - 1)} {self.processing_status}"
```

File: django_app/redbox_app/redbox_core/models.py [hunk 3]
```diff
@@ -31,3 +31,28 @@ class User(BaseUser, UUIDPrimaryKeyBase):
...
+    chunking = "chunking"
+    embedding = "embedding"
+    indexing = "indexing"
+    complete = "complete"
+
+
+# TO DO: Based on /redbox/models/file.py, but not complete
+class File(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    path = models.TextField()
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/models.py:48]:
```suggestion
    path = models.TextField(help_text="location of file")
```

**Review by gecBurton — APPROVED** [review]
added some some nitpicking but otherwise code is good.

File: django_app/redbox_app/redbox_core/migrations/0003_file.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 4.2.11 on 2024-03-20 08:18
+
+from django.db import migrations, models
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/migrations/0003_file.py:3]: note to self, auto linting doesnt appear to be working..... this is a problem for me to fix

**Comment by KevinEtchells** [comment]
Thanks George, appreciate the suggestions, which I'll work through now.

**Pull request closed without merging**